packages/crayons-core/src/components/country-phone/country-phone.tsx

```diff
@@ -0,0 +1,411 @@
+      }
+    }
+  }
+
+  private getMeta(isValid: boolean) {
+    return {
+      meta: {
+        isValid,
+        countryCode: this.countryCode,
+        countryName: this.countryName,
+        phoneCode: this.phoneCode,
+      },
+    };
+  }
+
+  private onInputBlur(event: Event) {
+    event.stopPropagation();
+    const value = (event.target as HTMLInputElement).value;
+    this.phoneNumber = value;
+    const isValid = isValidPhoneNumber(value, this.countryCode as CountryCode);
+    this.updateValue();
+    this.fwBlur.emit({
+      event,
+      name: this.name,
+      value: this.getSingleFormat(this.phoneCode, value),
+      ...this.getMeta(isValid),
+    });
+  }
+
+  private onInput(event: Event) {
+    event.stopPropagation();
+    const value = (event.target as HTMLInputElement).value;
+    this.phoneNumber = value;
+    const isValid = isValidPhoneNumber(value, this.countryCode as CountryCode);
+
+    this.updateValue();
+
+    this.fwInput.emit({
+      event,
+      name: this.name,
+      value: this.getSingleFormat(this.phoneCode, value),
+      ...this.getMeta(isValid),
+    });
+  }
+
+  private onInputClear(event: Event) {
+    event.stopPropagation();
+    const value = (event.target as HTMLInputElement).value;
+    this.updateValue();
+    this.fwInputClear.emit({
+      event,
+      name: this.name,
+      value: this.getSingleFormat(this.phoneCode, value),
+      ...this.getMeta(false),
+    });
+  }
+
+  private onInputFocus(event: Event) {
+    event.stopPropagation();
+    const value = (event.target as HTMLInputElement).value;
+    this.updateValue();
+    this.fwFocus.emit({
+      event,
+      name: this.name,
+      value: this.getSingleFormat(this.phoneCode, value),
+      ...this.getMeta(false),
+    });
+  }
+
+  private getCountryDetails(value: string) {
+    return countries.filter((r) => r.code2l === value);
+  }
+
+  private getCountryName(inputCountryCode: string) {
+    const currentCountry = this.getCountryDetails(inputCountryCode);
+    return (
+      currentCountry &&
+      currentCountry.length > 0 &&
+      currentCountry[0]?.country_name
+    );
+  }
+
+  private onSelectChange(event: Event) {
+    event.stopPropagation();
+    const value = (event.target as HTMLInputElement).value;
+    this.countryCode = value as CountryCode;
+
+    const currentCountry = this.getCountryDetails(value);
+    this.phoneCode = currentCountry[0].phone;
+    this.countryName = currentCountry[0].country_name;
+    this.phoneNumber = '';
+    this.isValueUpdatedFromInside = true;
+    this.updateValue();
+  }
+
+  render() {
+    /**
+     * @todo : Add new features
+     * 1. give option to include countries
+     * 2. give option to exclude countries
+     * */
+
+    const { host, name, value } = this;
+    renderHiddenField(host, name, value);
+
+    return (
+      <div class='country-container'>
+        <div
+          class='country-select-container'
+          style={{ flex: `0 0 ${this.countryWidth}px` }}
```

Comment:
should we use countryWidth to style the width of this element ?. or can we achieve it via css variables?

@Vignesh-Manogar-E3433 @vangakirankumarreddy

Reply:
I agree, CSS variables could be the better option here. Any styling customisation, we will expose as CSS variables rather than as props.

Reply:
Fixed ✅